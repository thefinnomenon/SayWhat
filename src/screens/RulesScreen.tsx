import React from 'react';
import { ScrollView } from 'react-native';
import styled from 'styled-components/native';
import { useTranslation } from 'react-i18next';
import normalize from '../../responsive';

export const RulesScreen = () => {
  const { t } = useTranslation();

  return (
    <Container>
      <ScrollView>
        <Line>
          Sit in a circle with every other person being on the same team.
        </Line>
        <Line>Decide which team is Team A and which is Team B.</Line>
        <Line>Choose the word deck you want to play with.</Line>
        <Line>A member of Team A should start the game</Line>
        <Line>
          The goal is to get your team members to guess the word or phrase that
          you see on the screen without using any of the words on the screen,
          using rhyming words, or spelling the words out.
        </Line>
        <Line>
          When one of your team members shouts the answer, hit the NEXT button
          and pass it to the player to your right.
        </Line>
        <Line>
          The team who is holding the device when the buzzer goes off, loses the
          round.
        </Line>
        <Line>Add a point to the winning team.</Line>
        <Line>First to 5 wins!</Line>
      </ScrollView>
    </Container>
  );
};

RulesScreen.navigationOptions = ({
  screenProps,
}: {
  screenProps: ScreenProps;
}) => ({
  title: screenProps.t('Rules'),
});

const Container = styled.View`
  flex: 1;
  background: ${props => props.theme.colors.background};
`;

const Line = styled.Text`
  color: ${props => props.theme.colors.text};
  font-size: ${normalize(20)};
  margin: ${normalize(10)}px;
`;

export default RulesScreen;
