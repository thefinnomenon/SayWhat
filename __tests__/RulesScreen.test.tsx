import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import RulesScreen from '../src/screens/RulesScreen';

it('renders correctly', () => {
  const tree = renderer.create(<RulesScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
