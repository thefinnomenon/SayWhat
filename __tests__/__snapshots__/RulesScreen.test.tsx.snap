// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <RCTScrollView>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#202020",
              "fontSize": 40,
              "marginBottom": 20,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 20,
            },
          ]
        }
      >
        Sit in a circle with every other person being on the same team.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#202020",
              "fontSize": 40,
              "marginBottom": 20,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 20,
            },
          ]
        }
      >
        Decide which team is Team A and which is Team B.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#202020",
              "fontSize": 40,
              "marginBottom": 20,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 20,
            },
          ]
        }
      >
        Choose the word deck you want to play with.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#202020",
              "fontSize": 40,
              "marginBottom": 20,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 20,
            },
          ]
        }
      >
        A member of Team A should start the game
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#202020",
              "fontSize": 40,
              "marginBottom": 20,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 20,
            },
          ]
        }
      >
        The goal is to get your team members to guess the word or phrase that you see on the screen without using any of the words on the screen, using rhyming words, or spelling the words out.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#202020",
              "fontSize": 40,
              "marginBottom": 20,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 20,
            },
          ]
        }
      >
        When one of your team members shouts the answer, hit the NEXT button and pass it to the player to your right.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#202020",
              "fontSize": 40,
              "marginBottom": 20,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 20,
            },
          ]
        }
      >
        The team who is holding the device when the buzzer goes off, loses the round.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#202020",
              "fontSize": 40,
              "marginBottom": 20,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 20,
            },
          ]
        }
      >
        Add a point to the winning team.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#202020",
              "fontSize": 40,
              "marginBottom": 20,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 20,
            },
          ]
        }
      >
        First to 5 wins!
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
